flash/text/classification/data.py

```diff
@@ -11,247 +11,168 @@
+        if input_key != DataKeys.INPUT:
+            hf_dataset = hf_dataset.rename_column(input_key, DataKeys.INPUT)
 
         return hf_dataset
 
-    def predict_load_data(self, data: Any, dataset: AutoDataset):
-        return self.load_data(data, dataset)
-
-    def __getstate__(self):  # TODO: Find out why this is being pickled
-        state = self.__dict__.copy()
-        state.pop("tokenizer")
-        return state
-
-    def __setstate__(self, state):
-        self.__dict__.update(state)
-        self.tokenizer = AutoTokenizer.from_pretrained(self.backbone, use_fast=True)
-
-
-class TextCSVInput(TextInput):
-    def to_hf_dataset(self, data: Tuple[str, str, str]) -> Tuple[Sequence[Mapping[str, Any]], str, str]:
-        file, *other = data
-        dataset_dict = load_dataset("csv", data_files={"train": str(file)})
-        return (dataset_dict["train"], *other)
-
-
-class TextJSONInput(TextInput):
-    def to_hf_dataset(self, data: Tuple[str, str, str, str]) -> Tuple[Sequence[Mapping[str, Any]], str, str]:
-        file, *other, field = data
-        dataset_dict = load_dataset("json", data_files={"train": str(file)}, field=field)
-        return (dataset_dict["train"], *other)
-
-
-class TextDataFrameInput(TextInput):
-    def to_hf_dataset(self, data: Tuple[DataFrame, str, str]) -> Tuple[Sequence[Mapping[str, Any]], str, str]:
-        df, *other = data
-        hf_dataset = Dataset.from_pandas(df)
-        return (hf_dataset, *other)
-
-
-class TextParquetInput(TextInput):
-    def to_hf_dataset(self, data: Tuple[str, str, str]) -> Tuple[Sequence[Mapping[str, Any]], str, str]:
-        file, *other = data
-        hf_dataset = Dataset.from_parquet(str(file))
-        return (hf_dataset, *other)
+    @property
+    @lru_cache(maxsize=None)
+    def tokenizer(self):
+        backbone_state = self.get_state(TextClassificationBackboneState)
+        return AutoTokenizer.from_pretrained(backbone_state.backbone, use_fast=True)
+
+    def load_sample(self, sample: Dict[str, Any]) -> Any:
+        tokenized_sample = self.tokenizer(
+            sample[DataKeys.INPUT], max_length=self.max_length, truncation=True, padding="max_length"
+        )
+        tokenized_sample = tokenized_sample.data
+        if DataKeys.TARGET in sample:
+            tokenized_sample[DataKeys.TARGET] = self.format_target(sample[DataKeys.TARGET])
+        return tokenized_sample
 
 
-class TextHuggingFaceDatasetInput(TextInput):
-    def to_hf_dataset(self, data: Tuple[str, str, str]) -> Tuple[Sequence[Mapping[str, Any]], str, str]:
-        hf_dataset, *other = data
-        return (hf_dataset, *other)
+class TextClassificationCSVInput(TextClassificationInput):
+    def load_data(
+        self,
+        csv_file: PATH_TYPE,
+        input_key: str,
+        target_keys: Optional[Union[str, List[str]]] = None,
+        max_length: int = 128,
+    ) -> Dataset:
+        dataset_dict = load_dataset("csv", data_files={"train": str(csv_file)})
```

Comment:
Would this work if we put a dummy key ? train might be confusing there.

Reply:
Yes, good catch :smiley: Actually, it looks like we don't need a key at all haha